grakn-graph/src/main/java/ai/grakn/graph/internal/ElementFactory.java

```diff
@@ -181,15 +184,35 @@ RoleTypeImpl buildRoleType(Vertex vertex, RoleType type, Boolean isImplicit){
                     concept = new RuleTypeImpl(graknGraph, v);
                     break;
                 default:
-                    throw new RuntimeException("Unknown base type");
+                    throw new RuntimeException("Unknown base type [" + v.label() + "]");
             }
             graknGraph.getConceptLog().cacheConcept(concept);
         }
 
         return graknGraph.getConceptLog().getCachedConcept(conceptId);
     }
 
-    EdgeImpl buildEdge(org.apache.tinkerpop.gremlin.structure.Edge edge, AbstractGraknGraph graknGraph){
+    //TODO: Simplify this if it does not make a difference to large loading
+    private Schema.BaseType getBaseType(Vertex vertex){
+        try {
+            return Schema.BaseType.valueOf(vertex.label());
+        } catch (IllegalArgumentException e){
+            //Base type appears to be invalid. Let's try getting the type via the isa edge
```

Comment:
So this will work for instances, but not types? I guess the main issue this is solving is ghosts when loading lots of instances, so it's fine.